### ISSUE TITLE
Google Drive authentication error: "Response code 404" when using GoogleDriveV0 gem

### ISSUE THREAD

**Issue by tdkehoe** [RailsApps/learn-rails#51]
In the chapter "Spreadsheet Connection," I'm getting an error message for the file app/models/contact.rb, at the line 

  connection = GoogleDriveV0.login(Rails.application.secrets.email_provider_username, Rails.application.secrets.email_provider_password)

I have my Gmail address and password in the file config/secrets.yml.

Any suggestions?
![screen shot 2015-03-05 at 1 08 04 pm](https://cloud.githubusercontent.com/assets/5279843/6513636/8a5c018a-c339-11e4-9075-255464801f46.png)

**Comment by DanielKehoe** [comment]
What's the error message? Is it an authentication failure?

Also see http://stackoverflow.com/questions/23414610/google-drive-authentication-error-in-rails

**Comment by DanielKehoe** [comment]
You were able to resolve this issue? Any tips for other readers who might encounter this?

**Comment by davefogo** [comment]
I am getting stuck at the same problem. I had to hardwire the credentials in the secrets.yml to bypass the error. Also tried the steps mentioned in the fore mentioned stack overflow  link. My password is capitalized, alpha-numerical and lacks any special chars. Any new updates?

**Comment by dxypher** [comment]
I'm getting stuck on this same line of code. Strangely I had it working yesterday but today it doesn't. Here is the new error I'm getting:

Response code 404 for post https://www.google.com/accounts/ClientLogin: https://developers.google.com/accounts/docs/AuthForInstalledApps

I don't have 2 step authentication for this email and 'Access to less secure apps' is set to allowed.

By the way, if you update the login credentials set in .bashrc or .zshrc, you may need to run the following command from the command line for those changes to work:

source ~/.bashrc
OR
source ~/.zshrc

**Comment by DanielKehoe** [comment]
I'm also seeing the error:
```
Started POST "/contacts" for ::1 at 2015-06-01 16:00:42 +0800
Processing by ContactsController#create as HTML
  Parameters: {...}
WARNING: GoogleDriveV0.login is deprecated and will be removed in the next version. Use GoogleDriveV0.login_with_oauth instead.
Completed 500 Internal Server Error in 1477ms (ActiveRecord: 0.0ms)

GoogleDriveV0::AuthenticationError - Authentication failed for fortuityconsulting@gmail.com: Response code 404 for post https://www.google.com/accounts/ClientLogin: https://developers.google.com/accounts/docs/AuthForInstalledApps
```
It looks like Google has replaced the authentication protocol (previously announced for April) as of May 26, 2015).

I'm updating the book for the new authentication protocol and will announce when the new edition is available.

**Comment by davefogo** [comment]
Thank you

**Comment by KONSTRUKTEUR** [comment]
I had the issues described above with Google Drive authentications and found out that I had to reload the .bashrc as described by dxypher. Once you edit your credentials into .bashrc you have to logout or you can use:

$ source ~/.bashrc

I am on OS-X and was not aware of this fact. The same problem would apply to all the other environmental variables we use in your tutorial. MAIL CHIMP, etc.

**Comment by avivalasvegas** [comment]
I am having this same error, does this mean I can' proceed forward with the tutorial, or should I use a different workaround?

**Comment by DanielKehoe** [comment]
Google now requires use of OAuth 2.0 for application access to Google Drive. The implementation is considerably more complex than the previous implementation using a Gmail address and password. I’ve released a new version of the book (version 2.2.0 now available on Amazon and online at learn-rails.com) and have dropped the “Spreadsheet Connection” chapter. I'm sure this will be disappointing but the chapter was primarily about adding a "fun feature" and doesn't diminish the value of the tutorial.
